src/openlcb/SimpleStack.hxx

```diff
@@ -456,6 +457,60 @@ private:
     std::unique_ptr<PhysicalIf> create_if(const openlcb::NodeID node_id);
 };
 
+class SimpleTcpStackBase : public SimpleStackBase
+{
+public:
+    SimpleTcpStackBase(const openlcb::NodeID node_id);
+
+    /// @return the device representing the connection to the TCP hardware
+    /// link.
+    HubFlow *tcp_hub()
+    {
+        return &static_cast<TcpPhysicalIf *>(ifaceHolder_.get())->tcpHub0_;
+    }
+
+protected:
+    /// Helper function for start_stack et al.
+    void start_iface(bool restart) override;
+
+private:
+    class TcpPhysicalIf : public PhysicalIf
+    {
+    public:
+        TcpPhysicalIf(const openlcb::NodeID node_id, Service *service)
+            : tcpHub0_(service)
+            , ifTcp_(node_id, &tcpHub0_, config_local_nodes_count())
+            , datagramService_(&ifTcp_, config_num_datagram_registry_entries(),
+                  config_num_datagram_clients())
+        {
+        }
+
+        ~TcpPhysicalIf()
+        {
+        }
+
+        /// @return the OpenLCB interface object. Ownership is not transferred.
+        If *iface() override
+        {
+            return &ifTcp_;
+        }
+        /// @return the Datagram service bound to the interface. Ownership is
+        /// not transferred.
+        DatagramService *datagram_service() override
+        {
+            return &datagramService_;
+        }
+        HubFlow tcpHub0_;
```

Comment:
What is the significance of the 0 at the end? Also since this gets assigned a Service object, is this the right name/type?

Reply:
0 is historical and I removed it now. Originally it was for CAN bus because some MCUs have two can-bus (can0 and can1).

The service object that it gets on the constructor is only a lightweight wrapper around the executor. It is only used to ensure that the flow runs on the main executor of the stack object.

Reply:
I figured it was related to that, but it makes sense to drop the zero here since there isn't really a hardware device backing the tcp stack (network interface excluded).

The note on the service object makes sense and explains the type as well.